test/OrchardCore.Tests/Apis/GraphQL/Queries/RecentBlogPostsQueryTests.cs

```diff
@@ -29,6 +29,10 @@ public async Task ShouldListBlogPostWhenCallingAQuery()
         // Indexing of the content item happens in the deferred-task and may not be immediate available,
         // so we wait until the indexing is done before querying.
         await context.WaitForOutstandingDeferredTasksAsync(TestContext.Current.CancellationToken);
+        await context.WaitForHttpBackgroundJobsAsync(TestContext.Current.CancellationToken);
+
+        // Indexing may still be running, so we wait a bit to ensure the content item is indexed.
+        await Task.Delay(2000, TestContext.Current.CancellationToken);
```

Comment:
@gvkries can you think of a better way than delaying?

Reply:
Understood, and I think this should work as-is, it's not clear to me why waiting will fail. If a deferred task starts the HTTP job with `ExecuteAfterEndOfRequestAsync`, `WaitForOutstandingDeferredTasksAsync` should complete, but `WaitForHttpBackgroundJobsAsync` must wait until the indexing has been completed.